### DISCUSSION TITLE
Code search vs Repos search

### DISCUSSION THREAD

**Discussion by sofiamigliorini** [community/community#64630] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Hi!
I need to retrieve images with architecture.png name in readme files
"architecture.png" path:/(^|\/)README\.md$/
this query works for code search but I need to get all 41k results and I don't find a good way to split the query, since created/update parameter are not allowed for search code.
I thought that 
"architecture.png" in:readme
will work for a repositories search with the same results but this isn't the case.
What's the error?

**Comment by EnriquezAlberto** [comment]
A python script

```
import requests

# Your GitHub token for authentication
token = "your_github_token_here"

# Base URL for the GitHub search API
base_url = "https://api.github.com/search/code"

# Search parameters
query = "architecture.png path:/(^|/)README.md$"
per_page = 100  # Number of results per page

# Initialize variables for pagination
page = 1
results = []

# Loop through pages of results
while True:
    params = {
        "q": query,
        "per_page": per_page,
        "page": page,
    }
    headers = {
        "Authorization": f"Bearer {token}",
    }
    
    response = requests.get(base_url, params=params, headers=headers)
    data = response.json()

    if "items" in data:
        results.extend(data["items"])

    # Check if there are more pages to retrieve
    if "next" in response.links:
        page += 1
    else:
        break

# Now, the 'results' list contains all the search results
print(f"Total results found: {len(results)}")
```

I hope I helped you bro :)

**Reply by sofiamigliorini** [reply]
Thank you for the pagination help, but it seems that this query doesn't give any result (even if I insert the escape characters) and I don't understand why :( I think the problem is the regex, even if in the GitHub search bar it works normally
